### PR TITLE
Refactor the specs to avoid using insist and in favor of a more rspec friendly syntax

### PR DESCRIPTION
**Pull request by purbon**

### CODE REVIEW

**Comment by guyboertje** [comment]
I have knocked up an alternative refactor that seeks to isolate the changing elements of the config and other variables to the example blocks so the reader can easily see the diffs across examples.

See [this gist](https://gist.github.com/guyboertje/12f7ec9e9ef1942145d7)

A discussion external to this PR should be had, to determine our expected level of DRY, brevity and readability of specs for plugins.

**Comment by purbon** [comment]
@guyboertje I agree in some parts of your refactor,  although I think we should find a mid term level between correctness, yours is more correct than mine in terms of DRY, and how easy are the specs to follow and understand for non ruby/rspec fluent people, even for some it could be more hard to follow.

I think this is a matter of a more in deep discussion we could drive for example opening a public issue in the core repo. 

In this PR review we should more focus on exact details, exact points to refactor to me more DRY, are the specs wrong? more test that should be added? what do you think?

**Comment by guyboertje** [comment]
Let agree to discuss this (DRY brevity vs readability) on another day, we need to for our Plugin Test guide doc anyway.

LGTM

**Comment by purbon** [comment]
@guyboertje agreed, feel free to open an issue about it, more than happy to jump in there.
